bots/dev/cli.py

```
@@ -1120,67 +1329,116 @@ def auto(self, bot: Bot, context: CLIContext, args: List[str]) -> str:
 
                     context.last_message_metrics = metrics.get_and_clear_last_metrics()
                 except Exception:
                     pass
 
-                # After sending prompt, check again if tools were used
-                if not bot.tool_handler.requests:
-                    restore_terminal(old_settings)
-                    return "Bot finished autonomous execution"
+                # Check for interrupt
+                if check_for_interrupt():
+                    raise KeyboardInterrupt("User interrupted")
```

Ruff (0.14.0)

Abstract `raise` to an inner function

(TRY301)